ros2_control_demo_hardware/src/rrbot_system_with_sensor.cpp

```diff
@@ -0,0 +1,245 @@
+// Copyright 2021 ros2_control Development Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ros2_control_demo_hardware/rrbot_system_with_sensor.hpp"
+
+#include <chrono>
+#include <cmath>
+#include <limits>
+#include <memory>
+#include <vector>
+
+#include "hardware_interface/types/hardware_interface_type_values.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+namespace ros2_control_demo_hardware
+{
+
+return_type RRBotSystemWithSensorHardware::configure(
+  const hardware_interface::HardwareInfo & info)
+{
+  if (configure_default(info) != return_type::OK) {
+    return return_type::ERROR;
+  }
+
+  hw_start_sec_ = stod(info_.hardware_parameters["example_param_hw_start_duration_sec"]);
+  hw_stop_sec_ = stod(info_.hardware_parameters["example_param_hw_stop_duration_sec"]);
+  hw_slowdown_ = stod(info_.hardware_parameters["example_param_hw_slowdown"]);
+  hw_sensor_change_ = stod(info_.hardware_parameters["example_param_max_sensor_change"]);
+  hw_joint_states_.resize(info_.joints.size(), std::numeric_limits<double>::quiet_NaN());
+  hw_joint_commands_.resize(info_.joints.size(), std::numeric_limits<double>::quiet_NaN());
+  hw_sensor_states_.resize(
+    info_.sensors[0].state_interfaces.size(),
+    std::numeric_limits<double>::quiet_NaN());
+
+  for (const hardware_interface::ComponentInfo & joint : info_.joints) {
+    // RRBotSystemWithSensor has exactly one state and command interface on each joint
+    if (joint.command_interfaces.size() != 1) {
+      RCLCPP_FATAL(
+        rclcpp::get_logger("RRBotSystemWithSensorHardware"),
```

Comment:
please put this string into a constant at the top of this file

Reply:
Good point. I think we have to also do this in other examples

Reply:
Replaced with constant string variable.